cmd/daserver/config.go

```diff
@@ -196,14 +230,72 @@ func (c *Config) GetSubmissionTimeout() (time.Duration, error) {
 	return time.ParseDuration(c.Submission.Timeout)
 }
 
-// GetReadTimeout parses and returns the read timeout duration.
+// GetReadTimeout parses and returns the read timeout duration for blob retrieval.
 func (c *Config) GetReadTimeout() (time.Duration, error) {
 	if c.Read.Timeout == "" {
 		return 30 * time.Second, nil
 	}
 	return time.ParseDuration(c.Read.Timeout)
 }
 
+// GetHTTPReadTimeout parses and returns the HTTP server read timeout.
+func (c *Config) GetHTTPReadTimeout() (time.Duration, error) {
+	if c.HTTPReadTimeout == "" {
+		return 30 * time.Second, nil
+	}
+	return time.ParseDuration(c.HTTPReadTimeout)
+}
+
+// GetHTTPWriteTimeout parses and returns the HTTP server write timeout.
+func (c *Config) GetHTTPWriteTimeout() (time.Duration, error) {
+	if c.HTTPWriteTimeout == "" {
+		return 120 * time.Second, nil
+	}
+	return time.ParseDuration(c.HTTPWriteTimeout)
+}
+
+// GetHTTPIdleTimeout parses and returns the HTTP server idle timeout.
+func (c *Config) GetHTTPIdleTimeout() (time.Duration, error) {
+	if c.HTTPIdleTimeout == "" {
+		return 60 * time.Second, nil
+	}
+	return time.ParseDuration(c.HTTPIdleTimeout)
+}
+
+// GetMaxBlobSize parses and returns the maximum blob size in bytes.
+func (c *Config) GetMaxBlobSize() (int64, error) {
+	if c.Submission.MaxBlobSize == "" {
+		return 2 * 1024 * 1024, nil // Default 2MB (Celestia max)
+	}
+	return parseByteSize(c.Submission.MaxBlobSize)
+}
+
+// parseByteSize parses human-readable byte sizes like "2MB", "1024KB", "2097152"
+func parseByteSize(s string) (int64, error) {
+	s = strings.TrimSpace(strings.ToUpper(s))
+
+	multipliers := map[string]int64{
+		"B":  1,
+		"KB": 1024,
+		"MB": 1024 * 1024,
+		"GB": 1024 * 1024 * 1024,
+	}
+
+	for suffix, mult := range multipliers {
+		if strings.HasSuffix(s, suffix) {
+			numStr := strings.TrimSuffix(s, suffix)
+			num, err := strconv.ParseInt(strings.TrimSpace(numStr), 10, 64)
+			if err != nil {
+				return 0, fmt.Errorf("invalid number in byte size: %w", err)
+			}
+			return num * mult, nil
+		}
+	}
+
+	// Plain number (bytes)
+	return strconv.ParseInt(s, 10, 64)
+}
+
```

Comment:
should use ordered slice

```suggestion
// parseByteSize parses human-readable byte sizes like "2MB", "1024KB", "2097152"
func parseByteSize(s string) (int64, error) {
	s = strings.TrimSpace(strings.ToUpper(s))

	units := []struct {
		suffix string
		mult   int64
	}{
		{"GB", 1024 * 1024 * 1024},
		{"MB", 1024 * 1024},
		{"KB", 1024},
		{"B", 1},
	}

	for _, u := range units {
		if strings.HasSuffix(s, u.suffix) {
			numStr := strings.TrimSuffix(s, u.suffix)
			num, err := strconv.ParseInt(strings.TrimSpace(numStr), 10, 64)
			if err != nil {
				return 0, fmt.Errorf("invalid number in byte size: %w", err)
			}
			return num * u.mult, nil
		}
	}

	// Plain number (bytes)
	return strconv.ParseInt(s, 10, 64)
}

```